### ISSUE TITLE
Most tests of  che-plugin-git-ext don't run any more

### ISSUE THREAD

**Issue by ChristianSchenk711** [eclipse-che/che#8705]
### Description

I'm building the che-plugin-git-ext-git plugin (with some modifications) and since we upgraded to 5.22.1 only 9 of 100+ test cases were executes.

In our maven test reports I just noticed that because the graph with the test cases dropped in number. There will not be any red tests, the tests are just not executed.

### Analysis

For me it looks like gwt-test-utils (0.47) throws an exception because it tries to access some classes which have been removed in mockito 2.10.0. (org.mockito.internal.configuration.DefaultAnnotationEngine)

**Comment by skabashnyuk** [comment]
Can you share exceptions or logs you have and steps to reproduce?

**Comment by ChristianSchenk711** [comment]
I just cloned the repo and ran mvn clean install (5.22.1) on che-plugin-git-ext-plugin on  a windows machine. But if you cannot confirm that behaviour, I will try again on a linux with a fresh setup. There are no exceptions just most of the tests are ignored. (all those that derive from BaseTests which uses the MockitoJunitRunner ).

**Comment by skabashnyuk** [comment]
Honestly, I haven't tried. @vinokurig  may be you remember if something was changed around ```che-plugin-git-ext-plugin```

**Comment by ChristianSchenk711** [comment]
Should be visible on the CI-server I guess. If there are only 9 tests executing in this plugin, a lot of them are not executed because there are 100+ in there.  When running e.g. CommitPresenterTest locally I get the exeception below. GwtInjectingAnnotationEngine is from gwt-utils-test 0.47 and uses DefaultAnnotationEngine which is not present any more in mockito 2.10.0.

```
java.lang.NoClassDefFoundError: org/mockito/internal/configuration/DefaultAnnotationEngine
	at org.mockito.configuration.GwtInjectingAnnotationEngine.<init>(GwtInjectingAnnotationEngine.java:30)
	at org.mockito.configuration.AnnotationEngineHolder.<clinit>(AnnotationEngineHolder.java:6)
	at org.mockito.configuration.MockitoConfiguration.getAnnotationEngine(MockitoConfiguration.java:47)
	at org.mockito.internal.configuration.GlobalConfiguration.tryGetPluginAnnotationEngine(GlobalConfiguration.java:57)
	at org.mockito.MockitoAnnotations.initMocks(MockitoAnnotations.java:68)
	at org.mockito.internal.runners.DefaultInternalRunner$1.withBefores(DefaultInternalRunner.java:38)
	at org.junit.runners.BlockJUnit4ClassRunner.methodBlock(BlockJUnit4ClassRunner.java:254)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.mockito.internal.runners.DefaultInternalRunner$1.run(DefaultInternalRunner.java:78)
	at org.mockito.internal.runners.DefaultInternalRunner.run(DefaultInternalRunner.java:84)
	at org.mockito.internal.runners.StrictRunner.run(StrictRunner.java:39)
	at org.mockito.junit.MockitoJUnitRunner.run(MockitoJUnitRunner.java:161)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:539)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:761)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:461)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:207)
```

**Comment by ChristianSchenk711** [comment]
Just tried again on a Ubuntu 64 bit VM outside the corporate network (no proxy whatever): Only 9 test cases run in the git-ext plugin, all others are ignored silently, reason most likely as described above. That's what I did:
```
git clone https://github.com/eclipse/che.git che
git checkout 5.22.1
git checkout -b 5221
cd che/plugins/plugin-git/che-plugin-git-ext-git
mvn clean install
```
I assume that was introduced when you switched to mockito 2.10.0 in #6394 .

**Comment by skabashnyuk** [comment]
AFAIK switch to mockito was introduced by @benoitf . Maybe he knows something.

**Comment by benoitf** [comment]
I will look at the issue

**Comment by benoitf** [comment]
so gwt-test-utils is not compliant with Mockito 2.x

I've tried to patch it so with the patch to use Mockito 2.x the tests are not silently ignored anymore
I could issue PR on gwt-test-utils but I don't know yet the timeframe (or if it will be accepted for merge)


With modified gwt-test-utils:
[ERROR] Tests run: 119, Failures: 26, Errors: 0, Skipped: 0

**Comment by eivantsov** [comment]
@ChristianSchenk711 is this still an issue for you?

**Comment by ChristianSchenk711** [comment]
Sorry for the late reply, was on leave. I just noticed because we added tests in this plugin and they vanished at some point. I have rewritten our tests to work now, so it's not an issue for me.

The key point here is that you probably also want your own tests to run nightly as well. But that's of course your decision (and I don't know your infrastructure, maybe those tests are visible for you?)
